tests/generator/test_generator_types.py

```diff
@@ -48,15 +47,49 @@ def test_make_generator_args(self):
 
         d = gen_args_as_program.first()
 
-        # First arguemnt: clvm deserializer
+        # First argument: clvm deserializer
 
-        b = hexstr_to_bytes("ff8568656c6c6fff86667269656e6480")  # ("hello" "friend")
+        b = bytes.fromhex("ff8568656c6c6fff86667269656e6480")  # ("hello" "friend")
         cost, output = d.run_with_cost([b])
         # print(cost, output)
         out = Program.to(output)
         assert out == Program.from_bytes(b)
 
         # Second Argument
-        arg2 = gen_args_as_program.rest().first().first()
+        arg2 = gen_args_as_program.rest().first()
         print(arg2)
         assert bytes(arg2) == bytes(gen1)
+
+    def test_list_to_tree(self):
+        self.assertEqual([], list_to_tree([]))
+        self.assertEqual(1, list_to_tree([1]))
+        self.assertEqual((1, 2), list_to_tree([1, 2]))
+        self.assertEqual(((1, 2), 3), list_to_tree([1, 2, 3]))
+        self.assertEqual(((1, 2), (3, 4)), list_to_tree([1, 2, 3, 4]))
+        self.assertEqual((((1, 2), 3), (4, 5)), list_to_tree([1, 2, 3, 4, 5]))
+        self.assertEqual((((1, 2), 3), ((4, 5), 6)), list_to_tree([1, 2, 3, 4, 5, 6]))
```

Comment:
so, each layer is not filled from left to right. I feel like that should be pointed out in a comment, and also in our documentation. This will be set in stone, right?

Reply:
Yes, argument setup will have to stay the same